projects/igniteui-angular/src/lib/date-common/util/date-time.util.ts

```diff
@@ -60,6 +60,10 @@ export abstract class DateTimeUtil {
             return null;
         }
 
+        if (/pm/g.test(inputData.toLowerCase())) {
```

Comment:
Won't work in all cases, specifically 12 AM which should end up as `0`. If the parts contain `ampm`  then all hour values are `mod 12` if you will, and theen you may add the 12 for PM as needed :)